alicat/serial.py

```diff
@@ -285,6 +439,99 @@ def set_pressure(self, pressure, retries=2):
             self._set_control_point('pressure', retries)
         self._set_setpoint(pressure, retries)
 
+    def hold(self, retries=2):
+        """Override command to issue a valve hold.
+
+        For a single valve mass flow/pressure controller, hold the valve at the present value.
+        For a dual valve mass flow controller, hold the valve at the present value.
+        For a dual valve pressure controller, close both valves.
+        """
+        self._test_controller_open()
+        command = '{addr}$$H\r'.format(addr=self.address)
+        self._write_and_read(command, retries)
+
+    def cancel_hold(self, retries=2):
+        """Cancel valve hold."""
+
+        self._test_controller_open()
+        command = '{addr}$$C\r'.format(addr=self.address)
+        self._write_and_read(command, retries)
+
+    def read_PID(self, retries=2):
+        """Read the current PID values on the controller.
+
+        Values include the loop type, P value, D value, and I value.
+
+        Values returned as a dictionary"""
+
+        self._test_controller_open()
+
+        self.pid_keys = ['loop_type', 'P', 'D', 'I']
+
+        command = '{addr}$$r85\r'.format(addr=self.address)
+        read_loop_type = self._write_and_read(command, retries)
+        spl = read_loop_type.split()
+
+        if spl[3] == '2':
+            loop_type = 'PD2I'
+        elif spl[3] == '1' or spl[3] == '0':
+            loop_type = 'PD/PDF'
```

Comment:
Nitpick here. I'd write as

```python
loop_type = ['PD', 'PDF', 'PD2I'].index(int(spl[3]))
```

This way, you'll get better errors when `spl[3]` is malformed.

Reply:
Readability here. Do in two lines even if one would work.

```python
loop_num = int(spl[3])
loop_type = ['PD/PDF', 'PD/PDF', 'PD2I'].index(loop_num)
```